test/widgets/action_sheet_test.dart

```diff
@@ -1188,6 +1189,199 @@ void main() {
       });
     });
   });
+
+  group('channel action sheet', () {
+    late ZulipStream someChannel;
+    const someTopic = 'my topic';
+    late StreamMessage someMessage;
+
+    Future<void> prepare({UnreadMessagesSnapshot? unreadMsgs}) async {
+      final stream = eg.stream();
+      someChannel = stream;
+      someMessage = eg.streamMessage(
+      stream: someChannel, topic: someTopic, sender: eg.otherUser);
+      addTearDown(testBinding.reset);
+
+      unreadMsgs ??= eg.unreadMsgs(channels: [
+        eg.unreadChannelMsgs(
+          streamId: stream.streamId,
+          topic: 'topic',
+          unreadMessageIds: [1],
```

Comment:
Is this meant to correspond to the same message as `someMessage`? If so it should use the same message ID.

In general, test data should be realistic. See discussion here:
[#frontend > typescript node tests @ 💬](https://chat.zulip.org/#narrow/channel/6-frontend/topic/typescript.20node.20tests/near/1896899)
Part of that is that the various different data structures in the model should be kept consistent with each other.